libfreerdp/core/connection.c

```diff
@@ -1172,6 +1170,18 @@ state_run_t rdp_client_connect_license(rdpRdp* rdp, wStream* s)
 			return STATE_RUN_FAILED;
 	}
 
+	/* there might be autodetect messages mixed in between licensing messages.
+	 * that has been observed with 2k12 R2 and 2k19
+	 */
+	const UINT16 messageChannelId = rdp->mcs->messageChannelId;
+	if (channelId == messageChannelId)
+	{
+		return rdp_recv_message_channel_pdu(rdp, s, securityFlags);
+	}
+
+	if (channelId != 1003)
```

Comment:
Not an expert with this code - but isn't 1003 the mcs channel id (`rdp->mcs->messageChannelId`) so that this check is obsolete (as we already checked for `==` above)?

Reply:
@mfleisz not necessarily (autodetect also uses `1008`), I´ve added this check so that this path is only used for the `1003` channel messages